core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -75,7 +77,7 @@ private[spark] class BlockManager(
 
   val diskBlockManager = new DiskBlockManager(this, conf)
 
-  private val blockInfo = new TimeStampedHashMap[BlockId, BlockInfo]
+  private val blockInfo = new ConcurrentHashMap[BlockId, BlockInfo]
```

Comment:
Isnt it simpler to keep the Scala map interface? Will minimize changes in rest of code in this class.